core/lib/include/compiler_hints.h

```diff
@@ -30,11 +30,13 @@ extern "C" {
  * @brief The *NORETURN* keyword tells the compiler to assume that the function
  *        cannot return.
  */
+#ifndef NORETURN
```

Comment:
This is unrelated to the commit message "define MAYBE_UNUSED"

Reply:
this contains https://github.com/RIOT-OS/RIOT/pull/18882

Reply:
Yeah, of course it's useful! But I recommend to split this off into its own commit ;-)